subsys/usb/class/netusb/eth_emu.c

```diff
@@ -0,0 +1,81 @@
+/* Ethernet Emulation driver
+ *
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+#define SYS_LOG_DOMAIN "dev/eth_emu"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+
+#include "netusb.h"
+#include "eth_emu.h"
+
+static struct eth_emu_context eth_emu_ctx;
+
+static int eth_emu_init(struct device *dev)
```

Comment:
eth_emu does not need to depend on usb, I think it would be better to separate it from this PR  and move to drivers/ethernet/, @jukkar your opinion?

Reply:
I can move it to some other place if needed.

Reply:
@jukkar @tbursztyka @rveerama1 could you check this virtual ethernet device?

Reply:
see also https://github.com/zephyrproject-rtos/zephyr/pull/4461#issuecomment-338216079